### PR TITLE
Balance Bot: updated SITL, deafult params

### PR DESCRIPTION
**Pull request by EbinPhilip**
- New physics simulator for SITL
- Modified default params
- Acro, Auto, RTL modes tested on SITL

### CODE REVIEW

**Comment by tridge** [comment]
@rmackay9 I think this is good for master. Note that with this PR the balance bot works in all modes (including AUTO missions) in SITL

File: libraries/SITL/SIM_BalanceBot.cpp [hunk 1]
```diff
@@ -52,53 +53,78 @@ float BalanceBot::calc_yaw_rate(float steering)
...
-    // a very crude model! Needs remodeling!
-    const float force_on_body = ((target_speed - velocity_vf_x) / max_speed) * max_force; //N
-
     // obtain roll, pitch, yaw from dcm
     float r, p, y;
     dcm.to_euler(&r, &p, &y);
-    float theta = p; //radians
-
-    float ang_vel = gyro.y; //radians/s
+//    float theta = p; //radians
```
Comment (rmackay9) [inline libraries/SITL/SIM_BalanceBot.cpp]: normally we don't leave commented out code in ..

Comment (rmackay9) [reply libraries/SITL/SIM_BalanceBot.cpp]: sorry, i see it's been deleted in the next commit..

**Comment by rmackay9** [comment]
It looks fine except one thing, can you just default ATC_BRAKE to "1" for all vehicles?  I've been meaning to make this change anyway.

**Comment by rmackay9** [comment]
Just one more change needed before this can go in.  The ATC_BRAKE default change should be done in AR_AttitudeControl.cpp instead of setting by name in APMrover2/Parameters.cpp.

**Comment by tridge** [comment]
looks like you've fixed everything @rmackay9 found, thanks!
merging

**Comment by rmackay9** [comment]
Yup, looks good!

**Pull request merged**